supertokens_python/recipe/emailpassword/api/utils.py

```diff
@@ -73,7 +73,7 @@ async def validate_form_fields_or_throw_error(
                 "All elements of formFields must contain an 'id' and 'value' field"
             )
         value = current_form_field["value"]
-        if current_form_field["id"] == FORM_FIELD_EMAIL_ID:
+        if current_form_field["id"] == FORM_FIELD_EMAIL_ID and isinstance(value, str):
```

Comment:
and if it's not a string type? Then should you allow it do go through? What happens in node SDK?

Reply:
> if it's not a string type?

then `value.strip()` won't be called but the `default_email_validator` will reject the value and return `"Email is not valid"`

> What happens in node SDK?
Same thing. It also seems vulnerable to the same bug.
```
formFields = formFields.map((field) => {
        if (field.id === FORM_FIELD_EMAIL_ID) {
            return {
                ...field,
                value: field.value.trim(),
            };
        }
        return field;
    });
```